src/tests/java/picard/analysis/CollectGcBiasMetricsTest.java

```diff
@@ -176,36 +124,193 @@ public void runTest(final File input) throws IOException {
+                "SUMMARY_OUTPUT=" + outfile.getAbsolutePath(),
+                "CHART_OUTPUT=" + pdf.getAbsolutePath(),
+                "SCAN_WINDOW_SIZE=" + windowSize,
+                "MINIMUM_GENOME_FRACTION=" + minGenFraction,
+                "IS_BISULFITE_SEQUENCED=" + biSulfiteSeq,
+                "LEVEL=ALL_READS",
+                "LEVEL=SAMPLE",
+                "LEVEL=READ_GROUP",
+                "ASSUME_SORTED=" + assumeSorted
+        };
+        Assert.assertEquals(runPicardCommandLine(args), 0);
+    }
+
+    /////////////////////////////////////////////////////////////////////////////
+    //Used to generate the Sam Record Sets with SamRecordSetBuilder.addPair().
+    //testNumber 1: runGcBiasMultiLevelTest, generates records aligning to chrM and chrO
+    //testNumber 2: runWindowsComparisonTest, generates records aligning to chrM,N,O.
+    /////////////////////////////////////////////////////////////////////////////
+    public void setup(final int ID, final String readGroupId, final SAMReadGroupRecord readGroupRecord, final String sample,
+                      final String library, final SAMFileHeader header, final SAMRecordSetBuilder setBuilder, final int testNumber)
+            throws IOException {
+
+        final String separator = ":";
+        final int contig1 = 0;
+        final int contig2 = 1;
+        final int contig3 = 2;
+        readGroupRecord.setSample(sample);
+        readGroupRecord.setPlatform(platform);
+        readGroupRecord.setLibrary(library);
+        readGroupRecord.setPlatformUnit(readGroupId);
+        if(testNumber==1) {header.addReadGroup(readGroupRecord);}
+        setBuilder.setReadGroup(readGroupRecord);
+        setBuilder.setUseNmFlag(true);
+
+        setBuilder.setHeader(header);
+
+        final int max = 800;
+        final int min = 1;
+        final Random rg = new Random(5);
+
+        //add records that align to chrM and O but not N
+        if(testNumber == 1) {
+            for (int i = 0; i < NUM_READS; i++) {
+                final int start = rg.nextInt(max) + min;
+                final String newReadName = READ_NAME + separator + ID + separator + i;
+
+                if (i != NUM_READS - 1) {
+                    setBuilder.addPair(newReadName, contig1, start + ID, start + ID + LENGTH);
+                } else {
+                    setBuilder.addPair(newReadName, contig2, start + ID, start + ID + LENGTH);
+                }
+            }
+        }
+
+        //add records that align to all 3 chr in reference file
+        else if(testNumber==2){
+            for (int i = 0; i < NUM_READS; i++) {
+                final int start = rg.nextInt(max) + min;
+                final String newReadName = READ_NAME + separator + ID + separator + i;
+
+                if (i<=NUM_READS/3) {
+                    setBuilder.addPair(newReadName, contig1, start + ID, start + ID + LENGTH);
+                } else if (i>NUM_READS/3 && i< (NUM_READS - (NUM_READS/3))) {
```

Comment:
The ```i>NUM_READS/3``` check is not needed here, since it's already ruled out by the previous clause.